### ISSUE TITLE
IP group JSON should support single addresses

### ISSUE THREAD

**Issue by pdurbin** [IQSS/dataverse#3340]
I gather from a336475 that IP groups should be able to be expressed as a single address. This should be tested/documented. I'm in the middle of resolving merge conflicts in pull request #3103 and I'm a little worried that I may be unfixing this? I'll leave a comment in a bit.

**Comment by michbarsinai** [comment]
re: the documentation: I'm note sure we have a central place for documenting the formats (we should, of course, but I'm not sure we have the personnel).
We do have example son files at `scripts/api/data/`, and there `ipGroup-localhost.json` does have this format.

**Comment by pdurbin** [comment]
@michbarsinai thanks for the fix at b0c0779 ! I just moved this card to the QA column at https://waffle.io/IQSS/dataverse

**Comment by kcondon** [comment]
This works but it requires that you specify both the ipv4 and ipv6 forms of the address, otherwise you get a 500 error. The range form allows only specifying ipv4, which is convenient.

Passing back to Michael for conisderation.

**Comment by kcondon** [comment]
Found another aspect to this issue: in addition to not allowing just ipv4, the ranges version only works with ipv4. It allows both but then does not properly detect the incoming ip addr.

**Comment by michbarsinai** [comment]
@kcondon I was not able to reproduce this (or, it's reverse day again). Here's what I did:
1. Added two new groups, one containing a single IPv4 address and one containing a single IPv6 one (`data/ipGroup-single-IPv4.json` and `data/ipGroup-single-IPv6.json`). Addition was done via API.
2. Added the localhost group (`data/ipGroup-localhost.json`).
3. Added the all-addresses group (`data/ipGroup-all.json`).

Up to this point - no 500 errors.

Then I ran inclusion tests using the test API endpoint `api/test/ipGroups/including/<address>`. I got correct inclusion results for the IPs in the groups.

**Comment by kcondon** [comment]
@michbarsinai 
All I did was 
1. try the localhost json but specify just my ip address, not localhost and only use the ipv4 part. It failed with an error.
2. I tried the ipgroup-all and that worked in creating the group but then when I assigned a role to it, my machine/user did not get the perms. When I removed  that group, then used the ipv4 version, perms were correctly granted, looking directly through ui, not using a test lib to check assignments
  
I can do a little more digging to more clearly identify steps if you want or if the above are not clear enough.
